src/nimble.nim

```diff
@@ -1025,23 +1027,23 @@ proc test(options: Options) =
   if not execHook(options, actionCustom, true):
     raise newException(NimbleError, "Pre-hook prevented further execution.")
 
-  files.sort((a, b) => cmp(a.path, b.path))
+  files.sort((a, b) => cmp(a, b))
+  files = files.map((a) => "tests" / a)
 
   for file in files:
-    let (_, name, ext) = file.path.splitFile()
-    if ext == ".nim" and name[0] == 't' and file.kind in {pcFile, pcLinkToFile}:
+    let (_, name, ext) = file.splitFile()
+    if ext == ".nim" and name[0] == 't':
       var optsCopy = options.briefClone()
       optsCopy.action = Action(typ: actionCompile)
-      optsCopy.action.file = file.path
+      optsCopy.action.file = file
       optsCopy.action.backend = pkgInfo.backend
       optsCopy.getCompilationFlags() = options.getCompilationFlags()
       # treat run flags as compile for default test task
       optsCopy.getCompilationFlags().add(options.action.custRunFlags)
       optsCopy.getCompilationFlags().add("-r")
+      optsCopy.getCompilationFlags().add("--out=" & ("build/" /
+          file.changeFileExt(ExeExt)))
```

Comment:
hm, don't we have a setting for this nowadays? @genotrance?

If not, maybe it would be better to put this in `tests/build/` or `tests/bin/`?

Reply:
Not sure what setting?

Related: https://github.com/nim-lang/nimble/issues/787

I think we should instead rely on `nim r` instead of worrying about yet another directory. That's only in devel though so we should wait a bit until it is in stable.

Reply:
My problem with relying on `nim r` is that it can leave executable files littered in folders, since in the original code you have to do a 'existedBefore' and 'existedAfter' check. If executable files, due to some crash or whatever reason, were not cleaned up properly, you now have these files scattered all over the test folder's subdirectories that will -not- get removed next 'nimble test' run (because existedBefore will now be true, so they won't get trashed).

Traditionally, a build folder is very clear that it's just build output and can be safely deleted.

I guess `tests/build` could work. 

I just think it's important to have this discussion because it would make `nimble test` plug and play.

My workflow is to mirror my 'tests' folder based on my src folder, more or less.
So if I had a `src/foo/foo_database.nim`, it would have a corresponding `tests/foo/test_foo_database.nim`